### DISCUSSION TITLE
What happens when there is a remainder in api.Div()

### DISCUSSION THREAD

**Discussion by sid-alluri** [Consensys/gnark#865] [General]
From what I [understood](https://github.com/Consensys/gnark/blob/59a4087261a6c73f13e80d695c17b398c3d0934f/frontend/cs/scs/api.go#L130), `api.Div()` constrains a = b * c when a/b = c. I am confused about how to perform a division that yields a remainder. For instance, when I perform the division of 6655/512, the output is a very big number. What does the output represent?

Code:
`api.Println(api.Div(frontend.Variable(6655), frontend.Variable(512)))`

Output:
`div_test.go:40 102413818701418340780171368180050714526739360352593042622272770898317541389 `

**Comment by sid-alluri** [comment]
Using lookup solved the issue for me.

**Reply by ivokub** [reply]
The operations are done modulo the scalar field modulus. See https://www.khanacademy.org/computing/computer-science/cryptography/modarithmetic/a/modular-inverses about how to compute modular inverse. Division is a multiplication of a modular inverse i.e. `a/b == a * (1/b)`.
